h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/task/XGBoostUploadMatrixTask.java

```diff
@@ -1,42 +1,375 @@
+            this.id = id;
+            this.rowHeader = new long[rowHeaderSize];
+            this.data = new float[dataSize];
+            this.colIndices = new int[dataSize];
+        }
+    }
+    
+    private class UploadSparseMatrixFun extends MrFun<UploadSparseMatrixFun> {
+
+        Frame _frame;
+        int[] _chunks;
+        Vec _weightVec;
+        Vec _offsetsVec;
+        DataInfo _di;
+        SparseMatrixDimensions _dims;
+        Vec _respVec;
+        float[] _resp;
+        float[] _weights;
+        float[] _offsets;
+
+        // OUT
+        int[] _actualRows;
+
+        UploadSparseMatrixFun(
+            Frame frame, int[] chunks, Vec weightVec, Vec offsetVec, DataInfo di,
+            SparseMatrixDimensions dimensions,
+            Vec respVec, float[] resp, float[] weights, float[] offsets
+        ) {
+            _actualRows = new int[chunks.length];
+
+            _frame = frame;
+            _chunks = chunks;
+            _weightVec = weightVec;
+            _offsetsVec = offsetVec;
+            _di = di;
+            _dims = dimensions;
+            _respVec = respVec;
+            _resp = resp;
+            _weights = weights;
+            _offsets = offsets;
+        }
+
+        @Override
+        protected void map(int chunkIdx) {
+            int chunk = _chunks[chunkIdx];
+            long nonZeroCount = _dims._precedingNonZeroElementsCounts[chunkIdx];
+            int rwRow = _dims._precedingRowCounts[chunkIdx];
+            int rowHeaderSize;
+            long dataSize;
+            if (chunkIdx == _dims._precedingNonZeroElementsCounts.length-1) {
+                rowHeaderSize = _dims._rowHeadersCount - rwRow;
+                dataSize = _dims._nonZeroElementsCount - nonZeroCount;
+            } else {
+                rowHeaderSize = _dims._precedingRowCounts[chunkIdx+1] - rwRow + 1;
+                dataSize = _dims._precedingNonZeroElementsCounts[chunkIdx+1] - nonZeroCount;
+            }
+            assert dataSize < Integer.MAX_VALUE;
+
+            Chunk weightChunk = _weightVec != null ? _weightVec.chunkForChunkIdx(chunk) : null;
+            Chunk offsetChunk = _offsetsVec != null ? _offsetsVec.chunkForChunkIdx(chunk) : null;
+            Chunk respChunk = _respVec.chunkForChunkIdx(chunk);
+            Chunk[] featChunks = new Chunk[_frame.vecs().length];
+            for (int i = 0; i < featChunks.length; i++) {
+                featChunks[i] = _frame.vecs()[i].chunkForChunkIdx(chunk);
+            }
+            SparseMatrixChunk chunkData = new SparseMatrixChunk(chunkIdx, rowHeaderSize, (int) dataSize);
+            int dataIndex = 0;
+            int rowHeaderIndex = 0;
+            for(int i = 0; i < respChunk._len; i++) {
+                if (weightChunk != null && weightChunk.atd(i) == 0) continue;
+                chunkData.rowHeader[rowHeaderIndex++] = nonZeroCount;
+                _actualRows[chunkIdx]++;
+                for (int j = 0; j < _di._cats; ++j) {
```

Comment:
it may sound pedantic, but even though there is no difference in ++j vs j++ in the for loop incrementation is there any reason why not to follow single convention? :)

Reply:
there is not, it should always be j++ as the convention dictates